compiler/rustc_expand/src/base.rs

```diff
@@ -1422,16 +1422,40 @@ fn pretty_printing_compatibility_hack(item: &Item, sess: &ParseSess) -> bool {
         if let ast::ItemKind::Enum(enum_def, _) = &item.kind {
             if let [variant] = &*enum_def.variants {
                 if variant.ident.name == sym::Input {
-                    sess.buffer_lint_with_diagnostic(
-                        &PROC_MACRO_BACK_COMPAT,
-                        item.ident.span,
-                        ast::CRATE_NODE_ID,
-                        "using `procedural-masquerade` crate",
-                        BuiltinLintDiagnostics::ProcMacroBackCompat(
-                        "The `procedural-masquerade` crate has been unnecessary since Rust 1.30.0. \
-                        Versions of this crate below 0.1.7 will eventually stop compiling.".to_string())
-                    );
-                    return true;
+                    let filename = sess.source_map().span_to_filename(item.ident.span);
```

Comment:
doesn't this only lint if `ProceduralMasqueradeDummyType::Input` is used in the `rental` crate, instead of - what i would expect - being from the `rental` crate but used in any other crate.

apart from this this change seems good to me, even though I am not too knowledgeable about this backcompat issue ^^ with this nit resolved, r=me after a compiler fcp

Reply:
That's correct - this type is an implementation detail of `rental`, and only gets used from a macro: https://github.com/jpernst/rental/blob/61188eb97f5d99b10fc1fdd469b259c4ab4bf940/src/lib.rs#L90-L98

Since it's in a macro, the file path will always be pointing to rental.